src/aquarium-optimized/d3d12/ContextD3D12.cpp

```diff
@@ -55,20 +55,24 @@ ContextD3D12::ContextD3D12()
     {
         mBufferSerias[n] = 0;
     }
+
+    mBackendName = "d3d12";
+    initDefaultToggleBitset();
 }
 
 ContextD3D12::~ContextD3D12() {}
 
-bool ContextD3D12::createContext(BACKENDTYPE backend, bool enableMSAA)
+bool ContextD3D12::initialize(BACKENDTYPE backend, const Aquarium &aquarium)
```

Comment:
If the only usage of aquarium is to call `aquarium.toggleBitset`, maybe you should pass toggleBitset instead of aquarium.